runtime/commonMain/src/kotlinx/serialization/modules/SerialModuleExtensions.kt

```diff
@@ -37,6 +52,12 @@ public fun <T : Any> SerialModule.getContextualOrDefault(klass: KClass<T>): KSer
  * @throws SerializationException if serializer can't be found.
  */
 @ImplicitReflectionSerializer
+@OptIn(UnsafeSerializationApi::class)
+@Deprecated(
+    level = DeprecationLevel.WARNING,
+    message = "This method is deprecated for removal. Please use reified getContextualOrDefault<T>() instead",
+    replaceWith = ReplaceWith("getContextual(value) ?: value::class.serializer()")
```

Comment:
we can actually replace with `getContextualOrDefault<T>()`, because we have value